sw/device/tests/sram_ctrl_execution_test.c

```diff
@@ -0,0 +1,51 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/dif/dif_sram_ctrl.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_isrs.h"
+#include "sw/device/lib/testing/test_framework/ottf_macros.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static dif_sram_ctrl_t sram_ctrl;
+
+enum {
+  kSramStart = TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR,
+  kSramEnd = TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR +
+             TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_SIZE_BYTES,
+  kSramRetStart = TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR,
+};
+
+OT_SECTION(".data.sram_function_test")
+OT_NOINLINE()
+static void sram_function_test(void) {
+  uint32_t pc = 0;
+  asm volatile("auipc %[pc], 0;" : [pc] "=r"(pc));
+  LOG_INFO("PC: %p, SRAM: [%p, %p)", pc, kSramStart, kSramEnd);
+  CHECK(pc >= kSramStart && pc < kSramEnd, "PC is outside the main SRAM");
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_sram_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_MAIN_REGS_BASE_ADDR),
+      &sram_ctrl));
+
+  // Note: We can test the negative case only using the retention SRAM since
+  // execution is unconditionally enabled for the main SRAM in the RMA life
+  // cycle state.
+  sram_ret_neg_test();
```

Comment:
It seems that this function will be defined only in the last commit right? Isn't it the case of squashing these two commits?

Reply:
Good catch, fixed.